tests/python/pants_test/init/test_extension_loader.py

```diff
@@ -365,7 +365,7 @@ def reg_alias():
 
     def test_subsystems(self):
         def global_subsystems():
-            return {DummySubsystem1, DummySubsystem2}
+            return [DummySubsystem1, DummySubsystem2]
```

Comment:
I just noticed this test fail in CI and reproduced locally. I'm fairly confident this is from earlier today switching us to use our own `OrderedSet`, rather than Twitter Commons. We implement `__eq__` more strictly than Twitter Commons, such as caring about it being the same type and the same order. 

Sets are inherently unstable, so I believe that sometimes this would getting resolved as `[Dummy1, Dummy2]` and sometimes `[Dummy2, Dummy1]`, thus causing a flaky test.

(Lmk if I should break this out into a dedicated PR)

Reply:
Is there a reason you switched it to a list instead of casting the return value to an unordered set for the purpose of the test comparison? Or are both of those perfectly good ways of solving the problem and you just picked one arbitrarily?

Reply:
I think that both are equally valid.